src/programmatic_policy_learning/envs/registry.py

```diff
@@ -0,0 +1,26 @@
+"""Environment provider registry."""
+
+from typing import Any, Callable, Dict
```

Comment:
general comment (that I still haven't gotten around to fixing in all my code): use `dict` instead of `Dict` to avoid the import. same goes for `list`, `tuple`, `set`. also I saw elsewhere `Optional` -- use `Foo | None` instead of `Optional[Foo]`